### PR TITLE
Fix/ Document roundings maxRedeem - issue 108

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes https://github.com/cantinasec/review-morpho-blue-1/issues/108#issuecomment-1791965416

### CODE REVIEW

File: src/MetaMorpho.sol [hunk 1]
```diff
@@ -472,11 +472,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     }
 
     /// @inheritdoc IERC4626
+    /// @notice Returns the max amount of assets a user can withdraw without reverting.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:475]: I think this one is redundant with `@inheritdoc IERC4626` which should already contains this description

**Review by Rubilmax — CHANGES_REQUESTED** [review]

File: src/MetaMorpho.sol [hunk 2]
```diff
@@ -472,11 +472,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
...
+    /// @notice Warning: because of roundings, this amount might be lower than the actual amount owned by the owner.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:476]:
```suggestion
    /// @dev Warning: may be lower than the actual amount of assets that can be withdrawn by `owner` due to conversion roundings between shares and assets.
```

File: src/MetaMorpho.sol [hunk 3]
```diff
@@ -472,11 +472,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
...
     function maxWithdraw(address owner) public view override(IERC4626, ERC4626) returns (uint256 assets) {
         (assets,,) = _maxWithdraw(owner);
     }
 
     /// @inheritdoc IERC4626
+    /// @notice Returns the max amount of shares a user can reddem without reverting.
+    /// @notice Warning: because of roundings, this amount might be lower than the actual amount owned by the owner.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:483]:
```suggestion
    /// @notice Warning: may be lower than the actual amount of shares that can be redeemed by `owner` due to conversion roundings between shares and assets.
```

**Review by Rubilmax — CHANGES_REQUESTED** [review]

File: src/MetaMorpho.sol [hunk 4]
```diff
@@ -472,11 +472,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     }
 
     /// @inheritdoc IERC4626
+    /// @dev Warning: may be lower than the actual amount of assets that can be withdrawn by `owner` due to conversion
```
Comment (MerlinEgalite) [inline src/MetaMorpho.sol:475]:
```suggestion
    /// @dev Warning: May be lower than the actual amount of assets that can be withdrawn by `owner` due to conversion
```

File: src/MetaMorpho.sol [hunk 5]
```diff
@@ -472,11 +472,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
...
+    /// roundings between shares and assets.
     function maxWithdraw(address owner) public view override(IERC4626, ERC4626) returns (uint256 assets) {
         (assets,,) = _maxWithdraw(owner);
     }
 
     /// @inheritdoc IERC4626
+    /// @dev Warning: may be lower than the actual amount of shares that can be redeemed by `owner` due to conversion
```
Comment (MerlinEgalite) [inline src/MetaMorpho.sol:482]:
```suggestion
    /// @dev Warning: May be lower than the actual amount of shares that can be redeemed by `owner` due to conversion
```

File: src/MetaMorpho.sol (hunk 4 above)
Comment (Rubilmax) [reply src/MetaMorpho.sol:475]: this indeed corresponds to how we wrote warnings in morpho-blue

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**